### PR TITLE
Feature/final touches before 0.8.0

### PR DESCRIPTION
**Pull request by kubaplas**
## Summary by CodeRabbit

- **Refactor**
	- Renamed entities for clarity and consistency: `ModuleWrapper` to `SubModule`, and `Variable` to `Var`.
- **Tests**
	- Ensured test suite accuracy post renaming of `Variable` to `Var` and `ModuleWrapper` to `SubModule.
- **Documentation**
	- Updated README and code comments to reflect recent changes for improved user and developer understanding.
- **Style**
	- Enhanced code readability and maintainability through adjustments in naming conventions and style.

### CODE REVIEW

**Comment by kubaplas** [comment]
Still missing:
- Remove odra-test from odra-vm

**Comment by kubaplas** [comment]
@kpob ready for review

File: Cargo.toml [hunk 1]
```diff
@@ -4,6 +4,7 @@ members = [
     "odra-vm",
     "core",
     "odra-macros",
+    "odra-casper/casper-client",
```
Comment (zie1ony) [inline Cargo.toml]: Why twice `odra-casper/casper-client`?
BTW, this name is a bit weird. Why don't we rename this crate to `odra-casper-rpc-client` in `odra-casper/rpc-client` directory?

File: core/src/variable.rs [hunk 2]
```diff
@@ -8,22 +8,22 @@ use crate::prelude::*;
 use crate::{OdraError, UnwrapOrRevert};
```
Comment (zie1ony) [inline core/src/variable.rs:8]: File name remains `variable.rs`?

File: examples/Cargo.toml [hunk 3]
```diff
@@ -4,7 +4,7 @@ version = "0.8.0"
 edition = "2021"
 
 [dependencies]
-odra = { path = "../odra", default-features = false }
+odra = { path = "../odra", version = "0.8.0", default-features = false }
```
Comment (zie1ony) [inline examples/Cargo.toml]: No need to do this in `examples`, this crate is not being published.

File: examples/src/features/collecting_events.rs [hunk 4]
```diff
@@ -3,8 +3,8 @@ use casper_event_standard::Event;
 use odra::casper_event_standard;
 use odra::prelude::*;
 use odra::{
-    module::{Module, ModuleWrapper},
-    Variable
+    module::{Module, SubModule},
```
Comment (zie1ony) [inline examples/src/features/collecting_events.rs]: Two things are wrong here.
1) `odra::module::Module` should not be imported. Code generated by macros use it, but it uses full path, so no need to import it. Please remove it from examples, and modules.
2) `SubModule` should be imported same way `Var` and `Mapping` so suggest it is `use odra::SubModule`.

File: examples/src/features/module_nesting.rs [hunk 5]
```diff
@@ -2,20 +2,20 @@ use casper_event_standard::Event;
 use odra::casper_event_standard;
```
Comment (zie1ony) [inline examples/src/features/module_nesting.rs:2]: I think this line might be avoided, if `casper_event_standard` is exported via `prelude`.

File: odra-vm/src/vm/odra_vm.rs [hunk 6]
```diff
@@ -616,7 +618,9 @@ mod tests {
     }
 
     fn test_caller(entry_point_name: &str) -> EntryPointsCaller {
-        let env = odra_test::env();
+        let vm = OdraVm::new();
+        let host_env = OdraVmHost::new(vm);
+        let env = HostEnv::new(host_env);
```
[the remark this answers is not in the document]
Comment (zie1ony) [reply odra-vm/src/vm/odra_vm.rs:622]: I think it solves the problem :)

File: templates/module.rs.template [hunk 7]
```diff
@@ -1,13 +1,13 @@
 use odra::prelude::*;
-use odra::Variable;
+use odra::Var;
 
-/// A module definition. Each module struct consists Variables and Mappings
+/// A module definition. Each module struct consists Vars and Mappings
```
Comment (zie1ony) [inline templates/module.rs.template:4]:
```suggestion
/// A module definition. Each module struct consists Vars and Mappings.
```

File: templates/module.rs.template [hunk 8]
```diff
@@ -36,12 +36,13 @@ impl #module_name {
...
 mod tests {
-    use super::#module_nameDeployer;
+    use super::*;
+    use odra::host::{Deployer, NoInit};
 
     #[test]
     fn it_works() {
         let env = odra_test::env();
-        let mut contract = #module_nameDeployer::init(&env);
+        let mut contract = #module_nameHostRef::deploy(&env, NoInit);
```
Comment (zie1ony) [inline templates/module.rs.template]:
```suggestion
        let mut contract = #module_nameHostRef::deploy(&env, NoArgs);
```
@kpob, first victim of `NoArgs` vs `NoInit`.

**Review by zie1ony — DISMISSED** [review]

File: examples/Cargo.toml (hunk 3 above)
Comment (kubaplas) [reply examples/Cargo.toml]: ok

File: templates/module.rs.template (hunk 8 above)
Comment (kpob) [reply templates/module.rs.template]: It's recommended to read the documentation first. :) 
My idea was `EmptyArgs` or `DefaultArgs`. Now both start with `No` that might be a source of confusion.

File: Cargo.toml (hunk 1 above)
Comment (kubaplas) [reply Cargo.toml]: Ok

File: core/src/variable.rs (hunk 2 above)
Comment (kubaplas) [reply core/src/variable.rs:8]: Renamed

File: examples/src/features/collecting_events.rs (hunk 4 above)
Comment (kubaplas) [reply examples/src/features/collecting_events.rs]: @zie1ony Doesn't work without importing Module:
```
error[E0599]: no method named `env` found for reference `&collecting_events::Machine` in the current scope
  --> src/features/collecting_events.rs:49:14
   |
49 |         self.env().emit_event(Info {
   |              ^^^ method not found in `&Machine`
   |
   = help: items from traits can only be used if the trait is in scope
help: some of the expressions' fields have a method of the same name
   |
49 |         self.e1.name.env().emit_event(Info {
   |              ++++++++
49 |         self.e2.name.env().emit_event(Info {
   |              ++++++++
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
   |
2  + use odra::module::Module;
   |
```

File: templates/module.rs.template (hunk 7 above)
Comment (kubaplas) [reply templates/module.rs.template:4]: There is a second part of this sentence line below

File: examples/src/features/collecting_events.rs (hunk 4 above)
Comment (kubaplas) [reply examples/src/features/collecting_events.rs]: I changed both:
```
use odra::{Submodule, Module}
```

File: templates/module.rs.template (hunk 8 above)
Comment (kubaplas) [reply templates/module.rs.template]: Fixed

File: core/src/lib.rs [hunk 9]
```diff
@@ -27,7 +27,7 @@ mod sequence;
 pub mod uints;
 mod unwrap_or_revert;
 pub mod utils;
-mod variable;
+mod var;
```
[the remark this answers is not in the document]
Comment (kubaplas) [reply core/src/lib.rs:30]: No

**Review by kpob — APPROVED** [review]

**Pull request merged**